browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -50,12 +90,36 @@ bool ApplyPotentialReferrerBlock(std::shared_ptr<BraveRequestInfo> ctx) {
   return false;
 }
 
+void ApplyPotentialQueryStringFilter(const GURL& request_url,
+                                     std::string* new_url_spec) {
+  std::string new_query = request_url.query();
```

Comment:
DCHECK(new_url_spec);

Reply:
please also add something like `SCOPED_UMA_HISTOGRAM_TIMER("Brave.SiteHacks.QueryFilter");` from `base/metrics/histogram_macros.h` so we can see the performance impact on `chrome://histograms`

Reply:
Done in latest revision.